### DISCUSSION TITLE
Batch Prediction Not working

### DISCUSSION THREAD

**Discussion by spatiallysaying** [obss/sahi#646] [Q&A] [answered]
I have followed the notebooks script for batch prediction of my custom Detectron2 segmentation model.

```
model_type = "detectron2"
model_path = '/content/output/model_final.pth'
model_config_path = model_path
model_device = "cpu" # or 'cuda:0'
model_confidence_threshold = 0.5

slice_height = 256
slice_width = 256
overlap_height_ratio = 0.2
overlap_width_ratio = 0.2

source_image_dir = "/content/Geotiff/"
```

```
predict(
    model_type=model_type,
    model_path=model_path,
    model_config_path=model_path,
    model_device=model_device,
    model_confidence_threshold=model_confidence_threshold,
    source=source_image_dir,
    slice_height=slice_height,
    slice_width=slice_width,
    overlap_height_ratio=overlap_height_ratio,
    overlap_width_ratio=overlap_width_ratio,
)
```

**Error Below**
There are 1 listed files in folder: Geotiff/
/content/output/model_final.pth not available in Model Zoo!

---------------------------------------------------------------------------

RuntimeError                              Traceback (most recent call last)

[/usr/local/lib/python3.7/dist-packages/sahi/model.py](https://localhost:8080/#) in load_model(self)
    521         try:  # try to load from model zoo
--> 522             config_file = model_zoo.get_config_file(self.config_path)
    523             cfg.merge_from_file(config_file)

12 frames

RuntimeError: /content/output/model_final.pth not available in Model Zoo!

During handling of the above exception, another exception occurred:

UnicodeDecodeError                        Traceback (most recent call last)

[/usr/lib/python3.7/codecs.py](https://localhost:8080/#) in decode(self, input, final)
    320         # decode input (taking the buffer into account)
    321         data = self.buffer + input
--> 322         (result, consumed) = self._buffer_decode(data, self.errors, final)
    323         # keep undecoded input until the next call
    324         self.buffer = data[consumed:]

UnicodeDecodeError: 'utf-8' codec can't decode byte 0x80 in position 64: invalid start byte

**Comment by fcakyon** [comment, marked as answer]
@spatiallysaying you cannot provide `model_config_path` as a model path. It has to be either a `.yaml file` or it should be `None`. 

Check the blogpost for mode detail on how do load Detectron2 models: https://medium.com/p/37063e33073f
